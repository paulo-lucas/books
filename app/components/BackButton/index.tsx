import React from 'react';
import { StyleSheet } from 'react-native';
import { Icon, useTheme } from '@rneui/themed';
import { useNavigation } from '@react-navigation/native';

export const BackButton = () => {
  const { theme } = useTheme();
  const navigation = useNavigation();

  return (
    <Icon
      name="arrowleft"
      type="antdesign"
      color={theme.colors.primary}
      onPress={navigation.goBack}
      size={24}
      style={styles.arrow}
    />
  );
};

const styles = StyleSheet.create({
  arrow: {
    marginRight: 20,
    padding: 10,
  },
});
