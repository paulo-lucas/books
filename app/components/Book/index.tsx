import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { Image, useTheme, Skeleton } from '@rneui/themed';
import FavoriteButton from './FavoriteButton';
import BookSkeleton from './BookSkeleton';
import { useNavigation, NavigationProp } from '@react-navigation/native';
import type { RootStackParamList } from '@app/router/rootStackParams';

interface BookProps {
  refreshing: boolean;
  title?: string;
  thumbnail?: string;
  authors?: Array<string>;
  publisher?: string;
  identifier?: string;
}

export const Book: React.FC<BookProps> = React.memo(
  ({ refreshing, title, thumbnail, authors, publisher, identifier }) => {
    const navigation = useNavigation<NavigationProp<RootStackParamList>>();
    const { theme } = useTheme();
    const styles = createStyles(theme);

    if (refreshing) {
      return <BookSkeleton />;
    }

    const author = authors?.length ? authors[0] : publisher;
    const image = thumbnail?.replace('http://', 'https://');
    const id = identifier as string;

    const goToDetails = () => navigation.navigate('Details', { id });

    return (
      <TouchableOpacity style={styles.touchable} onPress={goToDetails}>
        <FavoriteButton style={styles.favorite} identifier={id} />
        <View style={styles.container}>
          <View style={styles.shadow}>
            <Image
              source={{
                uri: image ?? 'https://www.xy.com/images/placeholder.jpg',
              }}
              containerStyle={styles.image}
              PlaceholderContent={<Skeleton width={300} height={500} />}
            />
          </View>
          <Text numberOfLines={2} style={styles.title}>
            {title}
          </Text>
          <Text numberOfLines={2} style={styles.author}>
            {author}
          </Text>
        </View>
      </TouchableOpacity>
    );
  },
);

const createStyles = ({ colors }: any) =>
  StyleSheet.create({
    touchable: {
      width: '50%',
      position: 'relative',
    },
    favorite: {
      position: 'absolute',
      right: 30,
      zIndex: 1,
    },
    container: {
      width: '100%',
      paddingHorizontal: 10,
      paddingBottom: 20,
      paddingTop: 5,
      alignItems: 'center',
      justifyContent: 'center',
    },
    image: {
      width: '100%',
      height: undefined,
      aspectRatio: 76 / 130,
      borderRadius: 5,
      elevation: 20,
    },
    shadow: {
      shadowOffset: { width: -2, height: 4 },
      shadowOpacity: 0.5,
      shadowRadius: 3,
    },
    title: {
      fontSize: 15,
      fontWeight: '500',
      paddingLeft: 5,
      textAlign: 'center',
      marginTop: 10,
      color: colors.black,
    },
    author: {
      fontSize: 13,
      paddingLeft: 5,
      textAlign: 'center',
      color: colors.black,
    },
  });
