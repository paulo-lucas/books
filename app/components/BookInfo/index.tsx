import React from 'react';
import { Text, View, ScrollView, StyleSheet } from 'react-native';
import { Image, Skeleton } from '@rneui/themed';
import { useBookDetails } from '@app/hooks';
import { getBestFromImageLinks } from '@app/utils/getBestImageFromImagelinks';
import BookInfoSkeleton from './BookInfoSkeleton';

export const BookInfo = () => {
  const { book, isFavorite, loading, toggleFavorite } = useBookDetails();

  if (loading) {
    return <BookInfoSkeleton />;
  }

  const image = getBestFromImageLinks(book?.volumeInfo.imageLinks);
  console.log(image);

  return (
    <ScrollView style={styles.container}>
      <View style={styles.imageWrapper}>
        <Image
          source={{ uri: image }}
          style={styles.image}
          placeholderStyle={styles.placeholder}
          PlaceholderContent={<Skeleton style={styles.placeholder} />}
        />
      </View>

      <Text style={styles.title}>{book?.volumeInfo.title}</Text>
      <Text style={styles.author}>{book?.volumeInfo.authors.join(', ')}</Text>
      <Text style={styles.description}>{book?.volumeInfo.description}</Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 40,
    paddingVertical: 20,
  },
  imageWrapper: {
    width: '100%',
    alignItems: 'center',
  },
  image: {
    width: '80%',
    height: undefined,
    aspectRatio: 76 / 130,
  },
  placeholder: {
    width: '100%',
    height: '100%',
  },
  title: {
    marginTop: 20,
    fontSize: 24,
    width: '85%',
  },
  author: {
    marginTop: 10,
    marginBottom: 20,
    fontSize: 18,
    width: '60%',
  },
  description: {
    marginTop: 5,
    fontSize: 20,
    width: '100%',
  },
});
