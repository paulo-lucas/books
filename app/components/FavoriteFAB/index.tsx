import React from 'react';
import { StyleSheet } from 'react-native';
import { FAB, useTheme } from '@rneui/themed';
import { useBookDetails } from '@app/hooks';

export const FavoriteFAB = () => {
  const { theme } = useTheme();
  const styles = createStyles(theme);
  const { isFavorite, toggleFavorite, loading } = useBookDetails();

  const iconColor = isFavorite() ? theme.colors.success : theme.colors.black;

  return (
    <FAB
      visible={!loading}
      buttonStyle={styles.button}
      onPress={toggleFavorite}
      icon={{ name: 'bookmark', type: 'ionicons', color: iconColor }}
      size="large"
      placement="right"
    />
  );
};

const createStyles = ({ colors }: any) =>
  StyleSheet.create({
    button: {
      backgroundColor: colors.surface,
    },
  });
