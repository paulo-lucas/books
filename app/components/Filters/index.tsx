import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import { SpeedDial, useTheme } from '@rneui/themed';
import { useFilters } from '@app/hooks';

export const Filters = () => {
  const { theme } = useTheme();
  const styles = createStyles(theme);

  const [open, setOpen] = useState<boolean>(false);
  const { toggleOrderBy, toggleFavoriteFilter, orderBy, filterByFavorites } =
    useFilters();

  const toggleOpen = () => setOpen(!open);

  return (
    <SpeedDial
      isOpen={open}
      buttonStyle={styles.primaryButton}
      icon={{
        name: 'filter',
        type: 'antdesign',
        color: '#fff',
      }}
      openIcon={{ name: 'close', color: '#fff' }}
      onOpen={toggleOpen}
      onClose={toggleOpen}>
      <SpeedDial.Action
        icon={{
          name: filterByFavorites ? 'bookmarks' : 'bookmark',
          type: 'ionicons',
          color: '#fff',
        }}
        title="Bookmarks"
        buttonStyle={
          filterByFavorites ? styles.secondaryButton : styles.primaryButton
        }
        onPress={toggleFavoriteFilter}
      />
      <SpeedDial.Action
        icon={{
          name: 'sort-ascending',
          type: 'material-community',
          color: '#fff',
        }}
        title={`Order by ${orderBy}`}
        buttonStyle={styles.primaryButton}
        onPress={toggleOrderBy}
      />
    </SpeedDial>
  );
};

const createStyles = ({ colors }: any) =>
  StyleSheet.create({
    primaryButton: {
      backgroundColor: colors.primary,
    },
    secondaryButton: {
      backgroundColor: colors.secondary,
    },
  });
