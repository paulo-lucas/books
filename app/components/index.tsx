export * from './ToggleTheme';
export * from './Bookshelf';
export * from './SearchBar';
export * from './Filters';
