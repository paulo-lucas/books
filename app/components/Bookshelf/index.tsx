import React from 'react';
import { FlatList } from 'react-native';
import { Volume } from '@app/interfaces/volume';
import { Book } from '@app/components';
import Empty from './Empty';
import { useBooks } from '@app/hooks';

interface BookshelfProps {
  volumes?: Array<Volume>;
}

interface RenderItemArgs {
  item: Volume;
}

export const Bookshelf: React.FC<BookshelfProps> = () => {
  const { books, fetchBooks, refreshing } = useBooks();

  const renderItem = ({ item }: RenderItemArgs) => {
    return (
      <Book
        refreshing={refreshing}
        title={item.volumeInfo?.title}
        publisher={item.volumeInfo?.publisher}
        thumbnail={item.volumeInfo?.imageLinks.thumbnail}
        authors={item.volumeInfo?.authors}
        identifier={item.id}
      />
    );
  };

  return (
    <FlatList
      refreshing={refreshing}
      data={
        refreshing ? ([{}, {}, {}, {}, {}, {}] as Array<Volume>) : books.data
      }
      renderItem={renderItem}
      keyExtractor={(_, id) => String(id)}
      ListEmptyComponent={Empty}
      numColumns={2}
      onRefresh={fetchBooks}
    />
  );
};
