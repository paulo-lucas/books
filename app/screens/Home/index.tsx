import React from 'react';
import { View, StyleSheet } from 'react-native';

import { Bookshelf, SearchBar, Filters, Paginator } from '@app/components';

function HomeScreen() {
  return (
    <>
      <View style={styles.container}>
        <SearchBar />
        <Paginator />
        <Bookshelf />
      </View>
      <Filters />
    </>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
});

export default HomeScreen;
