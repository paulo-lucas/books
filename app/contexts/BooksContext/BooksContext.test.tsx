import 'react-native';
import React, { ReactElement } from 'react';
import { render, RenderOptions, screen, act } from '../../../tests/test-utils';
import { BooksProvider } from './';
import { Bookshelf } from '@app/components';
import Home from '@app/screens/Home';
jest.useFakeTimers();

const customRender = (ui: ReactElement, options: RenderOptions = {}) => {
  return render(<BooksProvider>{ui}</BooksProvider>, options);
};

describe('books context', () => {
  it('should render correctly with consumer', async () => {
    customRender(<Home />);
    act(() => {
      jest.runAllTimers();
    });

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should render empty bookshelf', () => {
    customRender(<Bookshelf />);
    act(() => {
      jest.runAllTimers();
    });

    expect(screen.getByText('There are no books to show.')).toBeTruthy();
  });
});
