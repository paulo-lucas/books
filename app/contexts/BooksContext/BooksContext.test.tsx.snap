// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`books context should render correctly with consumer 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "transparent",
          "borderBottomWidth": 1,
          "borderTopColor": "transparent",
          "borderTopWidth": 1,
          "padding": 8,
          "paddingHorizontal": 15,
          "paddingVertical": 20,
        }
      }
      testID="RNE__SearchBar-wrapper"
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
        testID="RNE__Input__view-wrapper"
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 3,
              "flexDirection": "row",
              "minHeight": 30,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  Object {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#86939e",
                          "fontSize": 18,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="RNE__ICON__Component"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <TextInput
            autoCorrect={false}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Search books"
            placeholderTextColor="#86939e"
            platform="default"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#86939e",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 10,
                "minHeight": 40,
              }
            }
            testID="RNE__SearchBar"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginRight": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "height": 0,
              "margin": 0,
              "padding": 0,
            }
          }
        />
      </View>
    </View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 600,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                Object {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              replaceTheme={[Function]}
              testID="RNE__ICON__CONTAINER_ACTION"
              updateTheme={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#242424",
                        "fontSize": 36,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="RNE__ICON__Component"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#242424",
                "fontSize": 18,
                "fontWeight": "500",
                "marginTop": 10,
              }
            }
          >
            There are no books to show.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(36, 36, 36, 0.6)",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <RCTSafeAreaView
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          Object {
            "opacity": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 16,
                "marginRight": 24,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "color": "black",
                  "elevation": 2,
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                },
                undefined,
              ]
            }
          >
            Bookmarks
          </Text>
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 28,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 2,
                  },
                  Object {
                    "borderRadius": 28,
                    "elevation": 4,
                  },
                  false,
                ]
              }
              testID="RNE_BUTTON_WRAPPER"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#00DAC4",
                      "borderRadius": 2,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "padding": 8,
                      "paddingHorizontal": 10,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 40,
                        "marginHorizontal": 5,
                        "overflow": "hidden",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                    testID="RNE__ICON__CONTAINER"
                  >
                    <View
                      accessibilityRole="button"
                      android_ripple={
                        Object {
                          "borderless": false,
                          "color": "rgba(0, 0, 0, 0.3)",
                          "radius": -5,
                        }
                      }
                      disabled={false}
                      testID="RNE__ICON__CONTAINER_ACTION"
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        testID="RNE__ICON"
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": "#fff",
                                "fontSize": 24,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="RNE__ICON__Component"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          Object {
            "opacity": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 16,
                "marginRight": 24,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "color": "black",
                  "elevation": 2,
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                },
                undefined,
              ]
            }
          >
            Order by relevance
          </Text>
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 28,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 2,
                  },
                  Object {
                    "borderRadius": 28,
                    "elevation": 4,
                  },
                  false,
                ]
              }
              testID="RNE_BUTTON_WRAPPER"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#00DAC4",
                      "borderRadius": 2,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "padding": 8,
                      "paddingHorizontal": 10,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 40,
                        "marginHorizontal": 5,
                        "overflow": "hidden",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                    testID="RNE__ICON__CONTAINER"
                  >
                    <View
                      accessibilityRole="button"
                      android_ripple={
                        Object {
                          "borderless": false,
                          "color": "rgba(0, 0, 0, 0.3)",
                          "radius": -5,
                        }
                      }
                      disabled={false}
                      testID="RNE__ICON__CONTAINER_ACTION"
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        testID="RNE__ICON"
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": "#fff",
                                "fontSize": 24,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Material Design Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="RNE__ICON__Component"
                        >
                          󰒼
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 16,
            "marginTop": 0,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 2,
              },
              Object {
                "borderRadius": 28,
                "elevation": 4,
              },
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#00DAC4",
                  "borderRadius": 2,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 56,
                  "justifyContent": "center",
                  "padding": 16,
                  "paddingHorizontal": 10,
                  "width": 56,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 56,
                    "marginHorizontal": 5,
                    "overflow": "hidden",
                    "padding": 16,
                    "width": 56,
                  }
                }
                testID="RNE__ICON__CONTAINER"
              >
                <View
                  accessibilityRole="button"
                  android_ripple={
                    Object {
                      "borderless": false,
                      "color": "rgba(0, 0, 0, 0.3)",
                      "radius": -5,
                    }
                  }
                  disabled={false}
                  testID="RNE__ICON__CONTAINER_ACTION"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    testID="RNE__ICON"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="RNE__ICON__Component"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RNCSafeAreaView>
`;
